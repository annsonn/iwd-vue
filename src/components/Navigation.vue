<template>
  <nav id="nav">
    <ul id="main-nav">
      <li class="nav-item">
        <div>
          <router-link to="/">
            Software Development
          </router-link>
        </div>
        <ul class="sub-nav">
          <li>
            <router-link to="/">
              Our Services
            </router-link>
          </li>
          <li>
            <router-link to="/">
              Focus Areas
            </router-link>
          </li>
          <li>
            <router-link to="/">
              The Intelliware Approach
            </router-link>
          </li>
          <li>
            <router-link to="/">
              Success Stories
            </router-link>
          </li>
        </ul>
      </li>
      <li class="nav-item">
        <div>
          <router-link to="/about">
            Why Intelliware
          </router-link>
        </div>
        <ul class="sub-nav">
          <li>
            <router-link to="/">
              Our Agile Journey
            </router-link>
          </li>
          <li>
            <router-link to="/">
              Technology Leadership
            </router-link>
          </li>
          <li>
            <router-link to="/">
              Community Initiatives
            </router-link>
          </li>
          <li>
            <router-link to="/">
              Leadership Team
            </router-link>
          </li>
        </ul>
      </li>
    </ul>
  </nav>
</template>

<script>
export default {
  name: "Navigation",
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
a {
  text-decoration: none;
}
ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

#main-nav {
  display: flex;
  flex-direction: row;
}

.nav-item {
  flex: 1;
}

.nav-item .sub-nav {
  display: none;
}

.nav-item:hover .sub-nav {
  display: block;
}

.sub-nav {
  position: absolute;
}
</style>
