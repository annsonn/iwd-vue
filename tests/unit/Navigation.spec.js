import { shallowMount } from "@vue/test-utils";
import Navigation from "@/components/Navigation.vue";

describe("Navigation", () => {
  it("should render and match snapshot", () => {
    const wrapper = shallowMount(Navigation);
    expect(wrapper).toMatchSnapshot();
  });
});
