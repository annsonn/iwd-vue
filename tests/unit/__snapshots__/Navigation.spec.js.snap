// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render and match snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {
        "0": Object {},
      },
      "__props": Object {
        "0": Array [
          Object {},
          Array [],
        ],
      },
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <nav
        id="nav"
      >
        <ul
          id="main-nav"
        >
          <li
            class="nav-item"
          >
            <div>
              <router-link
                to="/"
              >
                 Software Development 
              </router-link>
            </div>
            <ul
              class="sub-nav"
            >
              <li>
                <router-link
                  to="/news"
                >
                   Our Services 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/news"
                >
                   Focus Areas 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/news"
                >
                   The Intelliware Approach 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/news"
                >
                   Success Stories 
                </router-link>
              </li>
            </ul>
          </li>
          <li
            class="nav-item"
          >
            <div>
              <router-link
                to="/news"
              >
                 Why Intelliware 
              </router-link>
            </div>
            <ul
              class="sub-nav"
            >
              <li>
                <router-link
                  to="/"
                >
                   Our Agile Journey 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/"
                >
                   Technology Leadership 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/"
                >
                   Community Initiatives 
                </router-link>
              </li>
              <li>
                <router-link
                  to="/"
                >
                   Leadership Team 
                </router-link>
              </li>
            </ul>
          </li>
        </ul>
      </nav>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
